internal/component/database_observability/postgres/collector/query_samples.go

```diff
@@ -258,22 +265,45 @@ func (c *QuerySamples) fetchQuerySample(ctx context.Context) error {
 			continue
 		}
 
-		c.upsertActiveSample(key, sample)
-		activeKeys[key] = struct{}{}
+		if !isIdleState(sample.State.String) {
+			c.upsertActiveSample(key, sample)
+			activeKeys[key] = struct{}{}
+			continue
+		}
+
+		if st, hadActive := c.samples[key]; hadActive {
+			end := sample.StateChange
+			if end.Valid {
+				st.EndOverride = end
+				st.LastSeenAt = end.Time
+			}
+			st.LastRow.State = sample.State
+			activeKeys[key] = struct{}{}
+		} else if _, already := c.emitted[key]; !already {
+			dummy := &SampleState{LastRow: sample, LastSeenAt: sample.Now, tracker: newWaitEventTracker()}
+			if sample.StateChange.Valid {
+				dummy.EndOverride = sample.StateChange
+				dummy.LastSeenAt = sample.StateChange.Time
+			}
+			c.samples[key] = dummy
+		}
+		// refresh emitted last seen for idle keys (dedupe & TTL)
+		c.emitted[key] = sample.Now
```

Comment:
Hmm probably this update needs to happen within the two IFs above. If we update unconditionally, there's a risk of edge case where an entry is idle in every single scrape and we continue refreshing the timestamp and thus never deleteing it from `c.emitted`. Does this resonate?